azurecompute-arm/src/main/java/org/jclouds/azurecomputearm/features/ResourceGroupApi.java

```
@@ -0,0 +1,99 @@
+import org.jclouds.azurecomputearm.domain.ResourceGroup;
+import org.jclouds.azurecomputearm.functions.StatusCodeParser;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.oauth.v2.filters.OAuthFilter;
+import org.jclouds.rest.annotations.QueryParams;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.SelectJson;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.MapBinder;
+import org.jclouds.rest.annotations.PayloadParam;
+import org.jclouds.rest.annotations.PATCH;
+import org.jclouds.rest.annotations.ResponseParser;
+
+
+import org.jclouds.rest.binders.BindToJsonPayload;
+
+/**
+ * The Azure Resource Manager API includes operations for managing resource groups in your subscription.
+ *
+ * @see <a href="https://msdn.microsoft.com/en-us/library/azure/dn790546.aspx">docs</a>
+ */
+@Path("/subscriptions/{subscriptionid}")
+
+@QueryParams(keys = "api-version", values = "2015-01-01")
+@RequestFilters(OAuthFilter.class)
+@Consumes(MediaType.APPLICATION_JSON)
+public interface ResourceGroupApi extends Closeable{
+
+   @Named("resourcegroup:list")
+   @Path("resourcegroups")
+   @SelectJson("value")
+   @GET
+   @Fallback(EmptyListOnNotFoundOr404.class)
+   List<ResourceGroup> list();
+
+   @Named("resourcegroup:create")
+   @PUT
+   @Path("resourcegroups/{name}")
+   @Produces(MediaType.APPLICATION_JSON)
+   @MapBinder(BindToJsonPayload.class)
+   ResourceGroup create(@PathParam("name") String name, @PayloadParam("location") String location, @PayloadParam("tags")Map<String, String> tags);
+
+   @Named("resourcegroup:get")
+   @GET
+   @Path("resourcegroups/{name}")
+   @Fallback(NullOnNotFoundOr404.class)
+   @Nullable
+   ResourceGroup get(@PathParam("name") String name);
+
+   @Named("resourcegroup:update")
+   @PATCH
+   @Produces(MediaType.APPLICATION_JSON)
+   @Path("resourcegroups/{name}")
+   @Fallback(NullOnNotFoundOr404.class)
```

We don't use 404 fallbacks on create/update operations, as they could mask the real failure. Just remove the fallback and let the method throw an exception if the referenced resource does not exist. Returning null in this case does not give the right feedback to users.